pkg/kubelet/kubelet.go

```diff
@@ -835,6 +835,10 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		StateDirectory:                   rootDirectory,
 	})
 	klet.shutdownManager = shutdownManager
+	klet.usernsManager, err = MakeUserNsManager(klet)
```

Comment:
is it better to protect creation of this object and references of this object with a feature gate?  right now, we always create the object, and therefore its used (or potentially used) when the feature gate is off.  it would seem safer to leave this field nil when feature is off or assign it a mock implementation.  thoughts?

Reply:
IMHO it is not needed, as all public methods start with the "if feature disabled" and it is a small number of methods (MakeUserNsManager, Release, GetUserNamespaceMappings, CleanupOrphanedPodUsernsAllocations, getHostIDsForPod).

But it seems like personal taste. If you prefer it that way, let me know and we can change it :)

Reply:
if its ok, i would prefer to leave the field nil if feature gate is not enabled.